Prog/get-zillow.R

```diff
@@ -0,0 +1,125 @@
+  select(-RegionType, -SizeRank, -RegionID, -StateName) %>% 
+  rename(Metro = RegionName) %>% 
+  gather(key = 'monthx', value = 'inventoryMetro', -Metro) %>% 
+  filter(Metro=="Washington, DC") %>% 
+  arrange(Metro, monthx) %>% 
+  mutate(month = substr(monthx, 6, 7),
+         year= substr(monthx, 1,4)) %>% 
+  filter(month=="07") %>% 
+  select(-monthx, -Metro, -month)
+
+#### inventory for COG region
+inventoryCOGJuly <- inventoryCOGmonth %>% 
+    filter(month=="07") %>% 
+    group_by(year) %>% 
+    summarize(JulyInventory= sum(inventory))
+
+
+# Sale prices --------------------------------------------------------
+
+price <- read_csv('http://files.zillowstatic.com/research/public/Metro/Sale_Prices_Msa.csv')
+
+priceMSAJuly <- price %>% 
+  select(-RegionID, -SizeRank) %>% 
+  rename(name = RegionName) %>% 
+  gather(key = 'monthx', value = 'Mediansaleprice', -name) %>% 
+  filter(name =="Washington, DC") %>%
+  mutate(month = substr(monthx, 6, 7), 
+         year= substr(monthx, 1,4)) %>% 
+  filter(month=="07") %>% 
+  select(-name, -monthx, -month)%>% 
+  arrange(year, Mediansaleprice, -month)
+
+# Rent for SF and MF residents --------------------------------------------------------
+
+rentSF <- read_csv('http://files.zillowstatic.com/research/public/Metro/Metro_MedianRentalPrice_Sfr.csv')
+rentlargeMF <- read_csv('http://files.zillowstatic.com/research/public/Metro/Metro_MedianRentalPrice_Mfr5Plus.csv')
+rentcondo <-read_csv('http://files.zillowstatic.com/research/public/Metro/Metro_MedianRentalPrice_CondoCoop.csv')
+rentduplex <-read_csv('http://files.zillowstatic.com/research/public/Metro/Metro_MedianRentalPrice_DuplexTriplex.csv')
+
+
+MetroRentSF <- rentSF %>% 
+  rename(Metro = RegionName) %>% 
+  select(-SizeRank) %>% 
+  filter(Metro=="Washington, DC") %>% 
+  gather(key = 'monthx', value = 'MedianSFRent', -Metro) %>% 
+  filter(Metro=="Washington, DC") %>% 
+  arrange(Metro, monthx) %>% 
+  mutate(month = substr(monthx, 6, 7),
+         year= substr(monthx, 1,4)) %>% 
+  filter(month=="07") %>% 
+  select(-monthx, -Metro, -month)
+
+
+MetroRentMF <- rentlargeMF %>% 
+  rename(Metro = RegionName) %>% 
+  select(-SizeRank) %>% 
+  filter(Metro=="Washington, DC") %>% 
+  gather(key = 'monthx', value = 'MedianMFRent', -Metro) %>% 
+  filter(Metro=="Washington, DC") %>% 
+  arrange(Metro, monthx) %>% 
+  mutate(month = substr(monthx, 6, 7),
+         year= substr(monthx, 1,4)) %>% 
+  filter(month=="07") %>% 
+  select(-monthx, -Metro, -month)
+
+MetroRentCondo <- rentcondo %>% 
+  rename(Metro = RegionName) %>% 
+  select(-SizeRank) %>% 
+  filter(Metro=="Washington, DC") %>% 
+  gather(key = 'monthx', value = 'MedianCondoRent', -Metro) %>% 
+  filter(Metro=="Washington, DC") %>% 
+  arrange(Metro, monthx) %>% 
+  mutate(month = substr(monthx, 6, 7),
+         year= substr(monthx, 1,4)) %>% 
+  filter(month=="07") %>% 
+  select(-monthx, -Metro, -month)
+
+MetroRentduplex <- rentduplex %>% 
+  rename(Metro = RegionName) %>% 
```

Comment:
all these statements run ok- but could be written as a function, with the series name and `value` as inputs. no need to rewrite, just noting.

Reply:
this would make it so you do not need to tweak each statement when you change something- just tweak the function, apply it so all the series.

Reply:
@sstrochak thanks, I wrote a function but I'm not sure how to only pass the dataframe name to be part of a new dataframe name, as well as how to define dataframe as function input. Could you take a look? The function is summedianprice.

Reply:
If you want to keep each series as a separate dataframe- and not convert to long and use `map_df()`- i would write like this, and run it for each series.  You can't use the assignment operator to assign objects to your global environment within a function.

```
get_monthly_price <- function(dataset, variable_name) {
  
  dataset %>% 
    rename(Metro = RegionName) %>% 
    select(-SizeRank) %>% 
    gather(key = 'monthx', value = variable_name, -Metro) %>% 
    filter(Metro=="Washington, DC") %>% 
    arrange(Metro, monthx) %>% 
    mutate(month = substr(monthx, 6, 7),
           year= substr(monthx, 1,4)) %>% 
    filter(month=="07") %>% 
    select(-monthx, -Metro, -month)
}


#From Zillow Rental Listings: Median rent list price
rentSF <- read_csv('http://files.zillowstatic.com/research/public/Metro/Metro_MedianRentalPrice_Sfr.csv')

MetroRentSF <- get_monthly_price(rentSF, "MedianSFRent")
```